test/pytest_3.10.patch

```diff
@@ -0,0 +1,38 @@
+diff --git a/_pytest/assertion/rewrite.py b/_pytest/assertion/rewrite.py
+index 4f96b9e8c..1aa5b12de 100644
+--- a/_pytest/assertion/rewrite.py
++++ b/_pytest/assertion/rewrite.py
+@@ -587,10 +587,6 @@ class AssertionRewriter(ast.NodeVisitor):
+             return
+         # Insert some special imports at the top of the module but after any
+         # docstrings and __future__ imports.
+-        aliases = [
+-            ast.alias(py.builtin.builtins.__name__, "@py_builtins"),
+-            ast.alias("_pytest.assertion.rewrite", "@pytest_ar"),
+-        ]
+         doc = getattr(mod, "docstring", None)
+         expect_docstring = doc is None
+         if doc is not None and self.is_rewrite_disabled(doc):
+@@ -617,6 +613,22 @@ class AssertionRewriter(ast.NodeVisitor):
+             pos += 1
+         else:
+             lineno = item.lineno
++        # Now actually insert the special imports.
```

Comment:
I don't quite understand this, but it seems it works because the tests pass. Would you mind instructing me on why lineno and col_offset are required for >3.10?

Reply:
This is a backport of this pytest PR: https://github.com/pytest-dev/pytest/pull/8540

it seems that `lineno` became a required argument to the function used here in Python 3.10.